.github/workflows/deploy-pnpm.yaml

```diff
@@ -0,0 +1,48 @@
+# Workflow intended to properly build the GitHub Pages Site. Using PNPM 
+
+name: Deployment Docs - PNPM 
+
+on:
+  # Trigger deployment on every push to main branch.
+  push:
+    branches: [main]
+  # Trigger deployment manually.
+  workflow_dispatch:
+    
+jobs:
+  deploy:
+    runs-on: ubuntu-latest 
+    
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # Fetch all commits to get last updated time or other git log info 
+          fetch-depth: 0
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '14'
+      - name: Setup PNPM 
+        run: npm install -g pnpm@v7.0.0
+      # Cache node_modules 
+      - name: Cache Dependencies 
+        uses: actions/cache@v2
+        id: pnpm-cache 
+        with:
+          path: |
+            **/node_modules 
+          key: ${{ runner.os }}-pnpm-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-pnpm-
+      - name: Install Dependencies 
+        if: steps.pnpm-cache.outputs.cache-hit != 'true'
+        run: pnpm install 
```

Comment:
I think this requires the `--frozen-lockfile` flag to behave like `npm ci` would

Reply:
I looked at a project of mine, where I used an older example from the pnpm documentation. They used two flags: `--frozen-lockfile` and `--strict-peer-dependencies`.